### PR TITLE
[EigenSolvers] Complex Solvers

### PR DESCRIPTION
**Pull request by oberbichler**
- Re-adding the complex solvers

It is compiling but I never tested the complex solvers. @qaumann maybe you can test them and give me some feedback.

### CODE REVIEW

File: kratos/includes/define_python.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 PYBIND11_DECLARE_HOLDER_TYPE(T, Kratos::intrusive_ptr<T>);
 
 #include <pybind11/stl.h>
+#include <pybind11/complex.h>
```
Comment (philbucher) [inline kratos/includes/define_python.h]: @qaumann this file is included almost everywhere in the py-interface
Is it possible to move this include to the files where you need it? I.e. to `add_matrix_to_python.cpp` and `add_vector_to_python.cpp`

Comment (qaumann) [reply kratos/includes/define_python.h]: Done.

**Comment by qaumann** [comment]
@philbucher this should be ready for review. In order to test the solvers in Python, I added a complex vector, matrix, and space to Python. The tests are provided.
The methods for sparse spaces are now templated (at least where it is meaningful) and some missing templates were added to matrix and vector.

**Comment by philbucher** [comment]
To me looks fine!
FYI @RiccardoRossi, maybe you also want to have a look...

Otherwise I would approve some time in the second week of january

File: kratos/python/add_matrix_to_python.cpp [hunk 2]
```diff
@@ -14,11 +14,12 @@
 // System includes
 
 // External includes
-
+#include "pybind11/complex.h"
```
Comment (oberbichler) [inline kratos/python/add_matrix_to_python.cpp]: Maybe use `<...>`

File: kratos/python/add_vector_to_python.cpp [hunk 3]
```diff
@@ -16,10 +16,12 @@
 // System includes
 
 // External includes
+#include "pybind11/complex.h"
```
Comment (oberbichler) [inline kratos/python/add_vector_to_python.cpp]: `<...>`

File: kratos/python/add_matrix_to_python.h [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 // System includes
 #include <pybind11/pybind11.h>
+#include <pybind11/complex.h>
```
Comment (philbucher) [inline kratos/python/add_matrix_to_python.h]: => cpp

File: kratos/python/add_vector_to_python.h [hunk 5]
```diff
@@ -20,6 +20,7 @@
 
 // System includes
 #include <pybind11/pybind11.h>
+#include <pybind11/complex.h>
```
Comment (philbucher) [inline kratos/python/add_vector_to_python.h]: => cpp

**Review by philbucher — COMMENTED** [review]
only minor comments, then ok for me

File: kratos/python/add_matrix_to_python.h (hunk 4 above)
Comment (qaumann) [reply kratos/python/add_matrix_to_python.h]: ok

**Review by philbucher — APPROVED** [review]
:+1:

approving since it does not change existing functionalities but rather fix some mistakes (e.g. `double` instead of `value_type`) and adds the support for complex things

**Comment by oberbichler** [comment]
Cool! Thx @qaumann @philbucher !

**Pull request merged**